Failures with astropy v4.0rc1

These are relevant looking failures. Please report back during the RC testing period whether they are issues  here only, or upstream in astropy. 

```
================================================================ test session starts ================================================================
platform darwin -- Python 3.7.4, pytest-5.2.1, py-1.8.0, pluggy-0.13.0

Running tests with ccdproc version 2.1.dev1562.
Running tests in ccdproc docs.

Date: 2019-11-20T15:40:39

Platform: Darwin-17.7.0-x86_64-i386-64bit

Executable: /usr/local/opt/python/bin/python3.7

Full Python Version: 
3.7.4 (default, Jul  9 2019, 18:15:00) 
[Clang 10.0.0 (clang-1000.11.45.5)]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 1.17.3
Scipy: 1.3.1
Matplotlib: 3.1.1.post2570+g70fb1a6b9
Pandas: 0.25.3
Astropy: 4.0rc1
astroscrappy: 1.0.8
reproject: 0.5.1
astropy-helpers: 4.0.rc1

Using Astropy options: remote_data: any.

rootdir: /private/var/folders/dc/hsm7tqpx2d57n7vb3k1l81xw0000gq/T/ccdproc-test-5_dbbnua/lib/python3.7/site-packages, inifile: setup.cfg
plugins: astropy-header-0.1.1, forked-1.0.2, arraydiff-0.3, dependency-0.4.0, xonsh-0.9.13.dev1, remotedata-0.3.2, openfiles-0.4.0, asdf-2.4.2, doctestplus-0.5.0, cov-2.8.1
collected 313 items                                                                                                                                 

ccdproc/combiner.py .                                                                                                                         [  0%]
ccdproc/core.py ....ss...                                                                                                                     [  3%]
ccdproc/extern/bitfield.py ..                                                                                                                 [  3%]
ccdproc/tests/test_bitfield.py ............                                                                                                   [  7%]
ccdproc/tests/test_ccdmask.py ...                                                                                                             [  8%]
ccdproc/tests/test_ccdproc.py ......................9
F.........................FFF....................                                         [ 31%]
ccdproc/tests/test_ccdproc_logging.py .......                                                                                                 [ 33%]
ccdproc/tests/test_combine_open_files.py ..                                                                                                   [ 34%]
ccdproc/tests/test_combiner.py ....................................................                                                           [ 50%]
ccdproc/tests/test_cosmicray.py ...............                                                                                               [ 55%]
ccdproc/tests/test_gain.py .....                                                                                                              [ 57%]
ccdproc/tests/test_image_collection.py ...........................................................................                            [ 81%]
ccdproc/tests/test_keyword.py ..........                                                                                                      [ 84%]
ccdproc/tests/test_memory_use.py sss                                                                                                          [ 85%]
ccdproc/tests/test_rebin.py ........                                                                                                          [ 87%]
ccdproc/tests/test_wrapped_external_funcs.py .F.                                                                                              [ 88%]
ccdproc/utils/slices.py .                                                                                                                     [ 89%]
ccdproc/utils/tests/test_slices.py ..................                                                                                         [ 94%]
docs/api.rst .                                                                                                                                [ 95%]
docs/authors_for_sphinx.rst .                                                                                                                 [ 95%]
docs/ccddata.rst .                                                                                                                            [ 95%]
docs/changelog.rst .                                                                                                                          [ 96%]
docs/citation.rst .                                                                                                                           [ 96%]
docs/conduct.rst .                                                                                                                            [ 96%]
docs/contributing.rst .                                                                                                                       [ 97%]
docs/getting_started.rst .                                                                                                                    [ 97%]
docs/image_combination.rst .                                                                                                                  [ 97%]
docs/image_management.rst .                                                                                                                   [ 98%]
docs/index.rst .                                                                                                                              [ 98%]
docs/install.rst .                                                                                                                            [ 98%]
docs/license.rst .                                                                                                                            [ 99%]
docs/overview.rst .                                                                                                                           [ 99%]
docs/reduction_examples.rst .                                                                                                                 [ 99%]
docs/reduction_toolbox.rst .                                                                                                                  [100%]

===================================================================== FAILURES ======================================================================
___________________________________________________________ test_trim_with_wcs_alters_wcs ___________________________________________________________

    def test_trim_with_wcs_alters_wcs():
        ccd_data = ccd_data_func()
        # WCS construction example pulled form astropy.wcs docs
        wcs = WCS(naxis=2)
        wcs.wcs.crpix = np.array(ccd_data.shape)/2
        wcs.wcs.cdelt = np.array([-0.066667, 0.066667])
        wcs.wcs.crval = [0, -90]
        wcs.wcs.ctype = ["RA---AIR", "DEC--AIR"]
        wcs.wcs.set_pv([(2, 1, 45.0)])
        ccd_wcs = CCDData(ccd_data, wcs=wcs)
        # The trim below should subtract 10 from the 2nd element of crpix.
        # (Second element because the FITS convention for index ordering is
        #  opposite that of python)
>       trimmed = trim_image(ccd_wcs[10:, :])

ccdproc/tests/test_ccdproc.py:313: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/Users/bsipocz/munka/devel/astropy/astropy/nddata/mixins/ndslicing.py:63: in __getitem__
    return self.__class__(**kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = CCDData([[-0.74882747,  0.56759473,  0.71815054, ...,  1.30935097,
           0.69587966,  0.32030821],
         [-0.6... -0.82190772],
         [-0.66845311, -2.15776154, -0.88186588, ..., -0.47226641,
           0.58196437,  0.97061286]])
args = ()
kwd = {'data': array([[-0.74882747,  0.56759473,  0.71815054, ...,  1.30935097,
         0.69587966,  0.32030821],
       [-...      0.58196437,  0.97061286]]), 'mask': None, 'meta': {'my_key': 42, 'your_key': 'not 42'}, 'uncertainty': None, ...}
llwcs = SlicedLowLevelWCS Transformation

This transformation has 2 pixel and 2 world dimensions

Array shape (Numpy order): N...elation between pixel and world axes:

           Pixel Dim
World Dim    0    1
        0  yes  yes
        1  yes  yes

    def __init__(self, *args, **kwd):
        if 'meta' not in kwd:
            kwd['meta'] = kwd.pop('header', None)
        if 'header' in kwd:
            raise ValueError("can't have both header and meta.")
    
        super().__init__(*args, **kwd)
        if self._wcs is not None:
            llwcs = self._wcs.low_level_wcs
            if not isinstance(llwcs, WCS):
>               raise TypeError("the wcs must be a WCS instance.")
E               TypeError: the wcs must be a WCS instance.

/Users/bsipocz/munka/devel/astropy/astropy/nddata/ccddata.py:194: TypeError
_________________________________________________________________ test_block_reduce _________________________________________________________________

    @pytest.mark.skipif(not HAS_BLOCK_X_FUNCS, reason="needs astropy >= 1.1.x")
    @pytest.mark.skipif((skimage.__version__ < '0.14.2') and
                        ('dev' in np.__version__),
                        reason="Incompatibility between scikit-image "
                               "and numpy 1.16")
    def test_block_reduce():
        ccd = CCDData(np.ones((4, 4)), unit='adu', meta={'testkw': 1},
                      mask=np.zeros((4, 4), dtype=bool),
                      uncertainty=StdDevUncertainty(np.ones((4, 4))),
>                     wcs=np.zeros((4, 4)))

ccdproc/tests/test_ccdproc.py:663: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = CCDData([[1., 1., 1., 1.],
         [1., 1., 1., 1.],
         [1., 1., 1., 1.],
         [1., 1., 1., 1.]])
args = (array([[1., 1., 1., 1.],
       [1., 1., 1., 1.],
       [1., 1., 1., 1.],
       [1., 1., 1., 1.]]),)
kwd = {'mask': array([[False, False, False, False],
       [False, False, False, False],
       [False, False, False, False]...      [1., 1., 1., 1.],
                   [1., 1., 1., 1.],
                   [1., 1., 1., 1.]]), 'unit': 'adu', ...}

    def __init__(self, *args, **kwd):
        if 'meta' not in kwd:
            kwd['meta'] = kwd.pop('header', None)
        if 'header' in kwd:
            raise ValueError("can't have both header and meta.")
    
        super().__init__(*args, **kwd)
        if self._wcs is not None:
>           llwcs = self._wcs.low_level_wcs
E           AttributeError: 'numpy.ndarray' object has no attribute 'low_level_wcs'

/Users/bsipocz/munka/devel/astropy/astropy/nddata/ccddata.py:192: AttributeError
________________________________________________________________ test_block_average _________________________________________________________________

    @pytest.mark.skipif(not HAS_BLOCK_X_FUNCS, reason="needs astropy >= 1.1.x")
    @pytest.mark.skipif((skimage.__version__ < '0.14.2') and
                        ('dev' in np.__version__),
                        reason="Incompatibility between scikit-image "
                               "and numpy 1.16")
    def test_block_average():
        ccd = CCDData(np.ones((4, 4)), unit='adu', meta={'testkw': 1},
                      mask=np.zeros((4, 4), dtype=bool),
                      uncertainty=StdDevUncertainty(np.ones((4, 4))),
>                     wcs=np.zeros((4, 4)))

ccdproc/tests/test_ccdproc.py:693: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = CCDData([[1., 1., 1., 1.],
         [1., 1., 1., 1.],
         [1., 1., 1., 1.],
         [1., 1., 1., 1.]])
args = (array([[1., 1., 1., 1.],
       [1., 1., 1., 1.],
       [1., 1., 1., 1.],
       [1., 1., 1., 1.]]),)
kwd = {'mask': array([[False, False, False, False],
       [False, False, False, False],
       [False, False, False, False]...      [1., 1., 1., 1.],
                   [1., 1., 1., 1.],
                   [1., 1., 1., 1.]]), 'unit': 'adu', ...}

    def __init__(self, *args, **kwd):
        if 'meta' not in kwd:
            kwd['meta'] = kwd.pop('header', None)
        if 'header' in kwd:
            raise ValueError("can't have both header and meta.")
    
        super().__init__(*args, **kwd)
        if self._wcs is not None:
>           llwcs = self._wcs.low_level_wcs
E           AttributeError: 'numpy.ndarray' object has no attribute 'low_level_wcs'

/Users/bsipocz/munka/devel/astropy/astropy/nddata/ccddata.py:192: AttributeError
_______________________________________________________________ test_block_replicate ________________________________________________________________

    @pytest.mark.skipif(not HAS_BLOCK_X_FUNCS, reason="needs astropy >= 1.1.x")
    def test_block_replicate():
        ccd = CCDData(np.ones((4, 4)), unit='adu', meta={'testkw': 1},
                      mask=np.zeros((4, 4), dtype=bool),
                      uncertainty=StdDevUncertainty(np.ones((4, 4))),
>                     wcs=np.zeros((4, 4)))

ccdproc/tests/test_ccdproc.py:721: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = CCDData([[1., 1., 1., 1.],
         [1., 1., 1., 1.],
         [1., 1., 1., 1.],
         [1., 1., 1., 1.]])
args = (array([[1., 1., 1., 1.],
       [1., 1., 1., 1.],
       [1., 1., 1., 1.],
       [1., 1., 1., 1.]]),)
kwd = {'mask': array([[False, False, False, False],
       [False, False, False, False],
       [False, False, False, False]...      [1., 1., 1., 1.],
                   [1., 1., 1., 1.],
                   [1., 1., 1., 1.]]), 'unit': 'adu', ...}

    def __init__(self, *args, **kwd):
        if 'meta' not in kwd:
            kwd['meta'] = kwd.pop('header', None)
        if 'header' in kwd:
            raise ValueError("can't have both header and meta.")
    
        super().__init__(*args, **kwd)
        if self._wcs is not None:
>           llwcs = self._wcs.low_level_wcs
E           AttributeError: 'numpy.ndarray' object has no attribute 'low_level_wcs'

/Users/bsipocz/munka/devel/astropy/astropy/nddata/ccddata.py:192: AttributeError
_____________________________________________________________ test_medianfilter_unusued _____________________________________________________________

    def test_medianfilter_unusued():
        ccd = CCDData(np.ones((3, 3)), unit='adu',
                      mask=np.ones((3, 3)),
                      uncertainty=StdDevUncertainty(np.ones((3, 3))),
                      wcs=np.ones((3, 3)),
>                     flags=np.ones((3, 3)))

../../ccdproc-test-5_dbbnua/lib/python3.7/site-packages/ccdproc/tests/test_wrapped_external_funcs.py:47: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = CCDData([[1., 1., 1.],
         [1., 1., 1.],
         [1., 1., 1.]])
args = (array([[1., 1., 1.],
       [1., 1., 1.],
       [1., 1., 1.]]),)
kwd = {'flags': array([[1., 1., 1.],
       [1., 1., 1.],
       [1., 1., 1.]]), 'mask': array([[1., 1., 1.],
       [1., 1....uncertainty': StdDevUncertainty([[1., 1., 1.],
                   [1., 1., 1.],
                   [1., 1., 1.]]), ...}

    def __init__(self, *args, **kwd):
        if 'meta' not in kwd:
            kwd['meta'] = kwd.pop('header', None)
        if 'header' in kwd:
            raise ValueError("can't have both header and meta.")
    
        super().__init__(*args, **kwd)
        if self._wcs is not None:
>           llwcs = self._wcs.low_level_wcs
E           AttributeError: 'numpy.ndarray' object has no attribute 'low_level_wcs'

/Users/bsipocz/munka/devel/astropy/astropy/nddata/ccddata.py:192: AttributeError
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -81,15 +81,6 @@ matrix:
                MAIN_CMD='pycodestyle ccdproc --count  --show-source --show-pep8'
                SETUP_CMD=''
 
-    allow_failures:
-        # Currently fails because of an issue in astropy, once it's fixed
-        # https://github.com/astropy/astropy/issues/9664 this test should not be
-        # allowed to fail
-        - env: PYTHON_VERSION=3.7
-               ASTROPY_VERSION=development
-               NUMPY_VERSION=stable
-               SETUP_CMD='test --coverage'
-
 install:
 
     - git clone git://github.com/astropy/ci-helpers.git

